### DISCUSSION TITLE
Where is the ball video coming from and how do I send video from my camera instead?

### DISCUSSION THREAD

**Discussion by noahweber1** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1891] [Q&A] [answered]
<img width="973" alt="123" src="https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/assets/39279932/bd8c77ca-63ce-4807-be7d-13c39d6a910a">
<img width="979" alt="1234" src="https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/assets/39279932/da8a9ca1-2fd5-4cac-828b-011e8cac7f51">

So I have indications that my connection (Viewer) works. I can get a stream, and the cloudwatch matric confirm it. HOWEVER, this "pingpong" is not what my camera is pointing at and these are the logs from the RPI where the ./kvsWebrtcClientMaster myChannel was executed.

<img width="632" alt="123456" src="https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/assets/39279932/a3bc804a-d792-4626-8296-8acf9c2f2a2b">

Hence, I dont know wether I am connected and If I am how to configure it to stream the actual camera contents.

I confirmed that gstreamer can capture the video feed (and upload to kinesis video ---> I understand this is different from a signaling channel!)

Question: How to stream a video on a RPI via webrtc

**Comment by sirknightj** [comment, marked as answer]
Hi, the ball moving around the screen pattern are sample frames from https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/tree/master/samples/h264SampleFrames. You need to use the master sample with gstreamer to pull the frames from the camera -- pass in `devicesrc` instead: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c?tab=readme-ov-file#sample-kvswebrtcclientmastergstsample

You can read more about that in the WebRTC workshop here
https://catalog.us-east-1.prod.workshops.aws/workshops/b95b9381-baf0-4bef-ba31-63817d54c2a6/en-US/6-webrtc-sdk/2-running

**Reply by sirknightj** [reply]
You may also want to check out this repo https://github.com/dave-malone/aws-kvs-webrtc-demo-for-raspberry-pi which has some utilities that might make setup on your raspberry pi easier.
